Core/APIRequest.swift

```diff
@@ -43,6 +44,10 @@ public class APIRequest {
         
     }
     
+    public enum APIRequestError: Error {
+        case noResponseOrError
+    }
```

Comment:
This exists since I wanted to add a way to use our `APIRequest` class with async/await, but really wanted to return a `Result` type with it instead of a tuple with two optionals. I did take a look at updating `APIRequestCompletion` to use a `Result` directly, but don't feel comfortable modifying a bunch of networking code as a part of this PR.